llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -1956,7 +1956,7 @@ maybeMoveCFI(MachineInstr &MI, MachineBasicBlock::iterator MaybeCFI) {
       MaybeCFI->getOpcode() != TargetOpcode::CFI_INSTRUCTION ||
       !(MI.getFlag(MachineInstr::FrameSetup) ||
         MI.getFlag(MachineInstr::FrameDestroy)) ||
-      AArch64InstrInfo::getLdStBaseOp(MI).getReg() != AArch64::SP)
+      MI.getOperand(0).getReg() != AArch64::SP)
```

Comment:
At first glance, this looks like a behavior change; AArch64InstrInfo::getLdStBaseOp returns operand 1 for an unrecognized instruction.  Am I missing something?

Reply:
We are calling this function with the "update" instruction, i.e. with something like `sub sp, sp, #n`, and want so see whether `SP` is modified. Using `getLdStBaseOp` is just non-sensical and used to work purely by accident.

Reply:
The only caller of `maybeMoveCFI()` is `AArch64LoadStoreOpt::mergeUpdateInsn()` where there is an `assert` to check that the update instruction is either `ADDXri` or `SUBXri`. Maybe repeat a similar `assert` in this function so that it checks its preconditions? This would make the code self-documenting and easier to read.

Reply:
Oh, they're the same value because of the context; that's fine, then.  Some assert/documentation indicating that MI is an update instruction would be helpful.

Reply:
Assertion added.